src/robot.cpp

```diff
@@ -154,7 +163,14 @@ void Robot::initialize(const std::string& name)
 
             if (shape)
             {
-                std::string full_link_name = "/" + name_ + "/" + link->name;
+                std::string full_link_name = tf_prefix_ + link->name;
+
+                // Don't prefix if link already starts with robot name
+                std::string entity_name;
+                if (link->name.substr(0, name_.length()) == name_)
```

Comment:
Which is better, check if link name starts with name, which can cause undesired behaviour if link starts with '/'. Which could be solved by using a find. (But might be slower, probably not significant)

Reply:
The problem here is that the hsrb description *does not* support (static) tf prefixing (in the xacro) and hence *not* in the resulting urdf on the parameter server and *not* in tf, while we *do* want this prefix in ED for easy filtering of WM entities. 

Look [here](http://wiki.ros.org/tf2/Migration#Removal_of_support_for_tf_prefix) for tf prefixing in tf 2

I'd propose to add this (a bit more clearly as a comment) since we don't have a proper solution at this moment.

Reply:
Comment added.